### PR TITLE
Feature/issue 299 button for clearing local data

### PR DESCRIPTION
**Pull request by rasmus-storjohann-PG**

### CODE REVIEW

**Comment by rasmus-storjohann-PG** [comment]
@bensinjin for review. Glad I got something working today!!

File: src/components/help/help_component.tsx [hunk 1]
```diff
@@ -34,46 +35,85 @@ const fixture: ReadonlyArray<HelpContact> = [
     },
 ];
 
-export const HelpComponent: React.StatelessComponent = (): JSX.Element => {
+export interface HelpComponentProps {
```
Comment (bensinjin) [inline src/components/help/help_component.tsx]: I've been thinking ... how would you feel about just having one type we export so we can mix actions and props together? It's a weird distinction we make between actions and props when they're all just props really. In this case we'd just export something called `type HelpComponentProps`. Do we gain something from having two separate interfaces?

File: src/components/help/help_component.tsx [hunk 2]
```diff
@@ -34,46 +35,85 @@ const fixture: ReadonlyArray<HelpContact> = [
...
+            'answers are chosen in the questionnaire, which tasks are in My Plan and which ' +
+            'tasks are marked as completed';
+
+        // tslint:disable-next-line:no-expression-statement
+        Alert.alert(alertHeading, alertMessage,
+            [
+                // tslint:disable-next-line:no-empty
+                { text: 'Cancel', onPress: (): void => { }, style: 'cancel' },
+                // tslint:disable-next-line:no-expression-statement
+                { text: 'Delete all user data', onPress: (): void => { props.clearAllUserState(); } },
```
Comment (bensinjin) [inline src/components/help/help_component.tsx]: I think this syntax: `onPress: (): void => props.clearAllUserState()` will remove the need to disable the ts rule.

File: src/components/help/help_connectede_component.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { clearAllUserData, ClearAllUserDataAction } from '../../stores/questionnaire/actions';
+
+const mapStateToProps = (_: Store): HelpComponentProps => ({
+});
+
+const mapDispatchToProps = (dispatch: Dispatch<Store>): HelpComponentActions => ({
+    clearAllUserState: (): ClearAllUserDataAction => dispatch(clearAllUserData()),
+});
+
+export const HelpConnectedComponent = connect(mapStateToProps, mapDispatchToProps)(HelpComponent);
```
Comment (bensinjin) [inline src/components/help/help_connectede_component.tsx]: Fix typo in this file's name

File: src/components/help/help_component.tsx (hunk 1 above)
Comment (rasmus-storjohann-PG) [reply src/components/help/help_component.tsx]: Hey, I'd love to do that, but the `connect()` API needs this separation so I don't see a way to do this cleanly.

File: src/components/help/help_component.tsx (hunk 2 above)
Comment (rasmus-storjohann-PG) [reply src/components/help/help_component.tsx]: Fixed

File: src/components/help/help_connectede_component.tsx (hunk 3 above)
Comment (rasmus-storjohann-PG) [reply src/components/help/help_connectede_component.tsx]: Fixed

**Comment by rasmus-storjohann-PG** [comment]
@bensinjin addressed your comments

File: src/components/help/help_component.tsx (hunk 1 above)
Comment (bensinjin) [reply src/components/help/help_component.tsx]: Oh right, yeah unfortunate

**Pull request merged**